e2e/test_operator/operator_test.go

```diff
@@ -124,6 +126,265 @@ var _ = Describe("Operator", Label("e2e"), func() {
+
+				Expect(*fdbCluster.GetAutomationOptions().Replacements.Enabled).To(BeTrue())
+
+			})
+
+			It("should eventually remove the targeted Pod whose Node is tainted", func() {
+				replacedPod := fixtures.RandomPickOnePod(initialPods.Items)
+				// Taint replacePod's node
+				node := fdbCluster.GetNode(replacedPod.Spec.NodeName)
+				node.Spec.Taints = []corev1.Taint{
+					{
+						Key:       taintKeyMaintenance,
+						Value:     "rack_maintenance",
+						Effect:    corev1.TaintEffectPreferNoSchedule,
+						TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Second * time.Duration(taintKeyMaintenanceDuration+1))},
+					},
+				}
+				log.Printf("Taint node: Pod name:%s Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", replacedPod.Name,
+					node.Name, node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+				fdbCluster.UpdateNode(node)
+
+				// Wait for operator to mark the node as tainted
+				// time.Sleep(time.Second * time.Duration(taintKeyStarDuration-taintKeyMaintenanceDuration))
+
+				// Wait for operator to replace the pod
+				// time.Sleep(time.Second * time.Duration(*fdbCluster.GetAutomationOptions().Replacements.TaintReplacementTimeSeconds+1))
+				err := fdbCluster.WaitForReconciliation()
+				log.Printf("Reconciliation error:%+v\n", err)
+				Expect(err).NotTo(HaveOccurred())
+
+				fdbCluster.EnsurePodIsDeletedWithCustomTimeout(replacedPod.Name, ensurePodIsDeletedTimeoutMinutes)
+			})
+
+			PIt("should eventually remove all Pods whose Nodes are tainted", func() {
+				replacedPods := fixtures.RandomPickPod(initialPods.Items, numNodesTainted)
+				for _, pod := range replacedPods {
+					// Taint replacePod's node
+					node := fdbCluster.GetNode(pod.Spec.NodeName)
+					node.Spec.Taints = []corev1.Taint{
+						{
+							Key:       taintKeyMaintenance,
+							Value:     "rack_maintenance",
+							Effect:    corev1.TaintEffectPreferNoSchedule,
+							TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Millisecond * time.Duration(taintKeyStarDuration*1000+int64(rand.Intn(1000))))},
+						},
+					}
+					log.Printf("Taint node: Pod name:%s Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", pod.Name,
+						node.Name, node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+					fdbCluster.UpdateNode(node)
+				}
+
+				// Wait for operator to replace the pod
+				// time.Sleep(time.Second * time.Duration(*fdbCluster.GetAutomationOptions().Replacements.TaintReplacementTimeSeconds+1))
+				err := fdbCluster.WaitForReconciliation()
+				log.Printf("Reconciliation error:%+v\n", err)
+				Expect(err).NotTo(HaveOccurred())
+
+				for i, pod := range replacedPods {
+					log.Printf("Ensure %dth pod:%s is deleted\n", i, pod.Name)
+					fdbCluster.EnsurePodIsDeletedWithCustomTimeout(pod.Name, ensurePodIsDeletedTimeoutMinutes)
+				}
+			})
+		})
+
+		When("cluster disables taint feature with star taint key", func() {
```

Comment:
Is it expected that this `When` statement is not part of the `Context` above? And shouldn't this be marked as `Pending` too?

Reply:
hmm, in my viscose, this When is inside Context above.